Engine/DataFeeds/ZipDataCacheProvider.cs

```diff
@@ -170,26 +160,80 @@ public void Dispose()
         /// <summary>
         /// Remove items in the cache that are older than the cutoff date
         /// </summary>
-        private void CleanCache()
+        private void CleanCache(DateTime utcNow)
         {
-            var clearCacheIfOlderThan = DateTime.UtcNow.AddSeconds(-CacheSeconds);
-
-            // clean all items that that are older than CacheSeconds than the current date
-            foreach (var zip in _zipFileCache)
+            // we just want one thread cleaning the cache at the same time
+            if (Monitor.TryEnter(_zipFileCache))
```

Comment:
Nice usage of `Monitor.TryEntry` here -- no point in everyone waiting, let the others skip 👍